src/main/resources/resource/Arduino/MrlComm/MrlComm.cpp

```diff
@@ -545,6 +544,7 @@ unsigned int MrlComm::getCustomMsg() {
  }
 
  void MrlComm::onDisconnect() {
+  msg->publishError("disconnected"); // mostly for debug purpose

```

Comment:
I don't think this will work.. if the serial port is disconnected.. you can't publish the error back.
i worry that this would break.. causing the following code to never be  executed.

Reply:
the onDisconnect was always called even if it was not disconnected. So I put that line to see if it's called at an inapropriate time.
The message will be put in the serial buffer just as many device will output to the serial buffer. I don't think it would break anything. But that line can be removed with no problem. it's just debugging thing